eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/OnlyFinalFieldsForImmutableClassCheck.java

```diff
@@ -0,0 +1,102 @@
+// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Verify the classes with annotation @Immutable should have following rules:
```

Comment:
Have you tried to render these javadocs? It should be `{@literal @Immutable}`. There are a few instances below.

Reply:
looks like adding @literal makes no difference, but, adding {@code @Immutable} makes it look better. What do you think

Reply:
also, I changed javadocs from private methods to simple comments.

Reply:
Odd that literal didn't _literally_ show up. *laugh* `@code` is fine.